src/wallet/wallet2.cpp

```diff
@@ -1559,6 +1559,23 @@ void wallet2::fast_refresh(uint64_t stop_height, uint64_t &blocks_start_height,
   }
 }
 
+
+bool wallet2::add_address_book_row(const cryptonote::account_public_address &address, const crypto::hash &payment_id, const std::string &description)
+{
+  wallet2::address_book_row a;
+  a.m_address = address;
+  a.m_payment_id = payment_id;
+  a.m_description = description;
+  
+  int key = (m_address_book.empty())? 0 : m_address_book.rbegin()->first;
+  bool r =  m_address_book.emplace(++key,a).second;
+  return r;
```

Comment:
That is very weird. If you don't care about ordering, and aren't actually indexing on a complex type, why have a map and those dummy keys which looks like they could go wrong when deleting, instead of, say, a deque ?